src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -714,10 +714,12 @@ public void unsetInvalidProxyHostAndPort() {
   public OCSPMode getOCSPMode() {
     OCSPMode ret;
 
+    Boolean disableOCSPMode =
+        (Boolean) connectionPropertiesMap.get(SFSessionProperty.DISABLE_OCSP_CHECKS);
     Boolean insecureMode = (Boolean) connectionPropertiesMap.get(SFSessionProperty.INSECURE_MODE);
-    if (insecureMode != null && insecureMode) {
+    if ((disableOCSPMode != null && disableOCSPMode) || (insecureMode != null && insecureMode)) {
```

Comment:
I don't think it works correctly in a case when someone deliberately specified `disableOCSPMode = false, insecureMode = true`, because the first `&&` will evaluate to false and we evaluate the second `&&` to `true` - and we shouldn't.

Reply:
this case should not happen - it's miss configuration
I think we can throw the exception when both disableOCSPMode and insecureMode are not null and are not equal

Reply:
guys didn't we call this `disableOCSPChecks` in other drivers, and in the requirements?

Reply:
got it, no need to change if it's just an internal variable name. i wanted to ensure `disableOCSPChecks` naming is consistent across all drivers